packages/api/src/WithViews.tsx

```diff
@@ -0,0 +1,30 @@
+import { RestViewDefinition } from '@syndesis/models';
+import * as React from 'react';
+import { DVFetch } from './DVFetch';
+import { IFetchState } from './Fetch';
+
+export interface IWithViewProps {
+  virtualizationId?: string;
+  initialValue?: RestViewDefinition[];
+  children(props: IFetchState<RestViewDefinition[]>): any;
+}
+
+/**
+ * A component that fetches the virtualization views with the specified identifier.
+ * @see [virtualizationId]{@link IWithViewProps#virtualizationId}
+ */
+export class WithViews extends React.Component<IWithViewProps> {
+  public render() {
+    return (
+      <DVFetch<RestViewDefinition[]>
+        url={`workspace/vdbs/${
+          this.props.virtualizationId
```

Comment:
`virtualizationId` could be undefined accordingly to the interface, which would generate an url like `workspace/vdbs/undefinedvdb/Models/views/Views`. Is this correct?

Reply:
Good catch @riccardo-forina. That should not be optional.